internal/infrastructure/storage/db/test/deposit_repository_impl_test.go

```diff
@@ -0,0 +1,103 @@
+package db
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+	"github.com/tdex-network/tdex-daemon/internal/infrastructure/storage/db/inmemory"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tdex-network/tdex-daemon/internal/core/domain"
+	dbbadger "github.com/tdex-network/tdex-daemon/internal/infrastructure/storage/db/badger"
+)
+
+func TestDepositRepositoryImplementations(t *testing.T) {
+	repositories := createDepositRepositories(t)
+
+	for i := range repositories {
+		repo := repositories[i]
+
+		t.Run(repo.Name, func(t *testing.T) {
+			t.Parallel()
+
+			t.Run("testAddAndListDeposits", func(t *testing.T) {
+				t.Parallel()
+				testAddAndListDeposits(t, repo)
+			})
+		})
+	}
+}
+
+func testAddAndListDeposits(t *testing.T, repo depositRepository) {
```

Comment:
we should check that attempting to add the same deposit/withdrawal more than once does not create copies into the repo impl (redundancy).

Reply:
I think there is no need to check this since [line](https://github.com/sekulicd/tdex-daemon/blob/d508c727f0e9fa49148fb2ffefc203f068de5428/internal/infrastructure/storage/db/badger/deposit_repository_impl.go#L69) prevents insertion for same key(same in withdrawals repo)

Reply:
these tests are not only for badger but for any impl of the repo domain interfaces. I have no doubt badger impl is fine but let's check if also the inmemory impl sticks with this constraint.

Reply:
agree

Reply:
Done